### PR TITLE
Introduce pseudo-code for problems

### PR DESCRIPTION
**Pull request by stwunsch**
We can try to identify exactly which loops we want to hide implicitely
by a vecops-like model.
Further, this shows nicely the number (or depth) of nested loops in common HEP
scenarios.

Which loops do we want to optimize/hide compared to code with fully explicit loops?

At which places in the workflow do we want to have a struct-like access to properties?

### CODE REVIEW

File: NNLOops.md [hunk 1]
```diff
@@ -24,10 +39,31 @@
...
+        closest_track = None
+        d = None
+        for t in tracks:
+            if t.pt > 1:
+                if d > (t.p - j.p)**2:
+                    d = (t.p - j.p)**2
+                    closest_track = t
+        hist_jpt.fill(j.pt)
+        hist_tpt.fill(closest_track.pt)
+        hist_dr.fill(sqrt(j.eta**2 + j.phi**2) - sqrt(closest_track.eta**2 + closest_track.phi**2))
```
Comment (eguiraud) [inline NNLOops.md]: could you add a newline so the table looks more compact? :D

**Comment by eguiraud** [comment]
thank you @stwunsch !

> Which loops do we want to optimize/hide compared to code with fully explicit loops?

All of them!

> At which places in the workflow do we want to have a struct-like access to properties?

Uhm hard to say a priori...

File: NNLOops.md (hunk 1 above)
Comment (stwunsch) [reply NNLOops.md]: added linebreaks and two-space indentations

**Comment by stwunsch** [comment]
I've compacted the table with line breaks and two spaces for indentations

**Comment by stwunsch** [comment]
> > Which loops do we want to optimize/hide compared to code with fully explicit loops?

> All of them!

Then let's write code with explicit loops and let us identify how we can condense it :)

**Pull request merged**